rskj-core/src/main/java/co/rsk/peg/BridgeSupport.java

```diff
@@ -1567,46 +1567,40 @@ private void adjustBalancesIfChangeOutputWasDust(BtcTransaction btcTx, Coin sent
      * The hash for the signature must be calculated with Transaction.SigHash.ALL and anyoneCanPay=false. The signature must be canonical.
      * If enough signatures were added, ask federators to broadcast the btc release tx.
      *
-     * @param federatorBtcPublicKey   Federator who is signing
-     * @param signatures              1 signature per btc tx input
-     * @param rskTxHash               The hash of the rsk tx
+     * @param federatorBtcPublicKey         Federator who is signing
+     * @param signatures                    1 signature per btc tx input
+     * @param releaseCreationRskTxHash      The hash of the release creation rsk tx
      */
-    public void addSignature(BtcECKey federatorBtcPublicKey, List<byte[]> signatures, Keccak256 rskTxHash) throws IOException {
+    public void addSignature(BtcECKey federatorBtcPublicKey, List<byte[]> signatures, Keccak256 releaseCreationRskTxHash) throws IOException {
         if (signatures == null || signatures.isEmpty()) {
             return;
         }
 
         Context.propagate(btcContext);
 
-        BtcTransaction releaseTx = provider.getPegoutsWaitingForSignatures().get(rskTxHash);
-        if (releaseTx == null) {
-            logger.warn("No tx waiting for signature for hash {}. Probably fully signed already.", rskTxHash);
-            return;
-        }
-        if (!hasEnoughSignatures(releaseTx, signatures)) {
+        if (svpIsOngoing() && isSvpSpendTx(releaseCreationRskTxHash)) {
+            addSvpSpendTxSignatures(federatorBtcPublicKey, signatures);
             return;
         }
 
-        addReleaseSignatures(federatorBtcPublicKey, signatures, rskTxHash, releaseTx);
-    }
-
-    private boolean hasEnoughSignatures(BtcTransaction releaseTx, List<byte[]> signatures) {
-        int inputsSize = releaseTx.getInputs().size();
-        int signaturesSize = signatures.size();
-
-        if (inputsSize != signaturesSize) {
-            logger.warn("Expected {} signatures but received {}.", inputsSize, signaturesSize);
-            return false;
-        }
-        return true;
+        addReleaseSignatures(federatorBtcPublicKey, signatures, releaseCreationRskTxHash);
     }
 
     private void addReleaseSignatures(
         BtcECKey federatorPublicKey,
         List<byte[]> signatures,
-        Keccak256 rskTxHash,
-        BtcTransaction btcTx
+        Keccak256 releaseCreationRskTxHash
     ) throws IOException {
+
+        BtcTransaction releaseTx = provider.getPegoutsWaitingForSignatures().get(releaseCreationRskTxHash);
+        if (releaseTx == null) {
+            logger.warn("No tx waiting for signature for hash {}. Probably fully signed already.", releaseCreationRskTxHash);
+            return;
+        }
+        if (!areSignaturesEnoughToSignAllTxInputs(releaseTx, signatures)) {
+            return;
```

Comment:
Maybe log this?

Reply:
is being logged :) 
https://github.com/rsksmart/rskj/blob/0bc0589d24b9667028b6c3b63b0b08ff74b3eca2/rskj-core/src/main/java/co/rsk/peg/BridgeSupport.java#L1699